nuxeo-drive-client/nxdrive/engine/engine.py

```diff
@@ -1005,3 +1009,29 @@ def fetch_all_users(self):
         except Exception as e:
             log.exception(e)
         return
+
+    def refresh_user(self, user_id):
+        '''
+            Save userid in user_cache 
+        '''
+        if not user_id:
+            return
+        if not hasattr(self, '_user_cache'):
+            log.trace("creating _user_cache ...")
+            self._user_cache = dict()
+        # If user_id is present in local cache, it is also present in database
+        # so return
+        if user_id in self._user_cache:
+            return
+        log.trace("user_id=%r not present in _user_cache" % user_id)
+        # Retrieve the user_info from local database
+        user_info = self._dao.get_user_info(user_id)
```

Comment:
Another interesting way to address this caching (instead of using _user_cache), if needed, is to use the lru_cache decorator (in sqlite module):
@functools.lru_cache()
def get_user_info(self, user_id)
  ...
This will cache query result (in an internal _dict_), known as [memoizing](https://en.wikipedia.org/wiki/Memoization) the function.